### PR TITLE
Linux update fix to send heartbeat data

### PR DESCRIPTION
**Pull request by jemex**
Today the agent doesn't send any data if the machine is not missing any available updates. This leads to uncertainty whether the machine is really up to date or it is missing updates but not sending data. with this change,the agent will send heartbeat data whether the machine is missing update or not.

### CODE REVIEW

File: test/code/plugins/patch_management_lib_test.rb [hunk 1]
```diff
@@ -257,17 +269,27 @@ def test_installed_packages_transform_and_wrap_with_installed_on_date
...
-        "Architecture"=>"all",
-        "PackageName" => "autotools-dev",
-        "PackageVersion" => "20150820.1",
-        "Repository" => nil,
-        "Size" => "151",
-        "Timestamp" => "2016-07-11T02:02:16.000Z"
-    }]
+    @myExpectedHash["DataItems"][0]["Collections"] = [
+                {
+                    "CollectionName" => "HeartBeatData_0.0.UpdateManagement.0_HeartBeat",
```
Comment (shuliums) [inline test/code/plugins/patch_management_lib_test.rb]: Please don't duplicate code. You can define your data as a class variable.

File: source/code/plugins/patch_management_lib.rb [hunk 2]
```diff
@@ -180,6 +180,19 @@ def availableUpdatesXMLtoHash(availableUpdatesXML, os_short_name)
         ret
     end
 
+    def availableHeartBeatData()
```
Comment (stasku) [inline source/code/plugins/patch_management_lib.rb]: here and everywhere - replace casing to be "Heartbeat"

File: source/code/plugins/patch_management_lib.rb [hunk 3]
```diff
@@ -180,6 +180,19 @@ def availableUpdatesXMLtoHash(availableUpdatesXML, os_short_name)
         ret
     end
 
+    def availableHeartBeatData()        
+        ret = {}
+        ret["CollectionName"] = "HeartBeatData" + @@delimiter + 
+                                "0.0.UpdateManagement.0"+ @@delimiter + "HeartBeat"
+        ret["PackageName"] = "UpdateManagementHeartBeat"
+        ret["Architecture"] = nil
```
Comment (stasku) [inline source/code/plugins/patch_management_lib.rb]: I just realized, if we do not set Architecture, complex key in ingestion will not work. Let's use value "all"

File: source/code/plugins/patch_management_lib.rb [hunk 4]
```diff
@@ -257,8 +270,8 @@ def transform_and_wrap(
         # Remove duplicate services because duplicate CollectionNames are not supported. TODO implement ordinal solution
         installedPackages = removeDuplicateCollectionNames(installedPackages)
         availableUpdates = removeDuplicateCollectionNames(availableUpdates)       
-        
-        collections = []
+        availableHeartBeat = availableHeartBeatData()
```
Comment (stasku) [inline source/code/plugins/patch_management_lib.rb]: rename to heartbeatItem

Comment (stasku) [inline source/code/plugins/patch_management_lib.rb]: and add the PR comment here to explain why we are adding the heartbeatItem

File: test/code/plugins/patch_management_lib_test.rb (hunk 1 above)
Comment (jemex) [inline test/code/plugins/patch_management_lib_test.rb]: Fixed

**Comment by lagalbra** [comment]
@Microsoft/omsagent-devs

File: source/code/plugins/patch_management_lib.rb [hunk 5]
```diff
@@ -199,7 +199,7 @@ def installedPackageXMLtoHash(packageXML, os_short_name)
         
         ret["CollectionName"] = packageHash["Name"] + @@delimiter + 
                                 packageHash["Version"] + @@delimiter + os_short_name
-        ret["PackageName"] = packageHash["Name"]
+        ret["PackageName"] = packageHash["Name"]
```
Comment (stasku) [inline source/code/plugins/patch_management_lib.rb]: did you add some extra spaces?

File: test/code/plugins/patch_management_lib_test.rb [hunk 6]
```diff
@@ -251,7 +251,7 @@ def test_installed_packages_transform_and_wrap
                 "Installed" => true,
                 "UpdateState"=>"NotNeeded",
                 "Architecture"=>"all",
-                "PackageName" => "autotools-dev",
+                "PackageName" => "autotools-dev",
```
Comment (stasku) [inline test/code/plugins/patch_management_lib_test.rb]: redundant spaces?

**Comment by lagalbra** [comment]
Please squash all your commits into one before merging this change into the master branch.

File: test/code/plugins/patch_management_lib_test.rb (hunk 6 above)
Comment (jemex) [inline test/code/plugins/patch_management_lib_test.rb]: Fixed

File: source/code/plugins/patch_management_lib.rb (hunk 5 above)
Comment (jemex) [inline source/code/plugins/patch_management_lib.rb]: Fixed

**Pull request merged**